hcx-scheduler-jobs/common-scheduler-job/src/main/java/org/swasth/commonscheduler/schedulers/CertificateRevocationScheduler.java

```diff
@@ -0,0 +1,120 @@
+package org.swasth.commonscheduler.schedulers;
+
+import kong.unirest.HttpResponse;
+import org.bouncycastle.cert.ocsp.OCSPException;
+import org.bouncycastle.operator.OperatorCreationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.swasth.common.exception.ClientException;
+import org.swasth.common.exception.ErrorCodes;
+import org.swasth.common.service.RegistryService;
+import org.swasth.common.utils.Constants;
+import org.swasth.common.utils.JSONUtils;
+import org.swasth.common.utils.NotificationUtils;
+import org.swasth.common.validation.CertificateRevocation;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.security.cert.CRLException;
+import java.security.cert.CertificateException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.List;
+import java.util.Map;
+
+@Component
+public class CertificateRevocationScheduler extends BaseScheduler {
+
+    private final Logger logger = LoggerFactory.getLogger(ParticipantValidationScheduler.class);
+
+    @Autowired
+    private RegistryService registryService;
+
+    @Value("${hcx.participantCode}")
+    private String hcxParticipantCode;
+
+    @Value("${hcx.privateKey}")
+    private String hcxPrivateKey;
+
+    @Value("${kafka.topic.notification}")
+    private String notifyTopic;
+
+    @Value("${topicCode.revoked-certificate}")
+    private String certificateRevokedTopicCode;
+    @Value("${topicCode.invalid-certificate}")
+    private String invalidCertificateTopicCode;
+
+    public void process() throws Exception {
+        logger.info("Certificate Revocation validation scheduler started");
+        List<Map<String, Object>> participants = registryService.getDetails("{ \"filters\": {} }");
+        List<String> revokedParticipantCodes = new ArrayList<>();
+        List<String> invalidCertificates = new ArrayList<>();
```

Comment:
Avoid using a mutable list like this. It creates problem and won't be able to understand what are all the places the list is updating.